### ISSUE TITLE
[double_must_use] False positives from macro expansion?

### ISSUE THREAD

**Issue by djc** [rust-lang/rust-clippy#17529]
### Summary

In https://github.com/rustls/rustls/actions/runs/31314382101/job/93247145466?pr=3200 we're seeing a bunch of `double_must_use` warnings that seems like false positives. I think these originate in macro-expanded code, which should probably be ignored?

### Lint Name

double_must_use

### Reproducer

I tried this code:

```rust
    #[async_trait(?Send)]
    pub(crate) trait AsyncRead {
        async fn read(&mut self, buf: &mut [u8]) -> io::Result<usize>;
        async fn read_exact(&mut self, buf: &mut [u8]) -> io::Result<()>;
    }
```

I saw this happen:

```rust
error: this function has a `#[must_use]` attribute with no message, but returns a type already considered as `#[must_use]`
   --> ci-bench/src/util.rs:103:26
    |
103 |       #[async_trait(?Send)]
    |  __________________________^
104 | |     pub(crate) trait AsyncRead {
105 | |         async fn read(&mut self, buf: &mut [u8]) -> io::Result<usize>;
    | |________^
    |
note: the return type is pinned boxed `std::future::Future` trait object
   --> ci-bench/src/util.rs:103:5
    |
103 |     #[async_trait(?Send)]
    |     ^^^^^^^^^^^^^^^^^^^^^
help: remove `must_use`
   --> ci-bench/src/util.rs:103:5
    |
103 |     #[async_trait(?Send)]
    |     ^^^^^^^^^^^^^^^^^^^^^
    = note: alternatively, you may add an explicit reason to the `must_use` attribute
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#double_must_use
    = note: `-D clippy::double-must-use` implied by `-D warnings`
    = help: to override `-D warnings` add `#[allow(clippy::double_must_use)]`
    = note: this error originates in the attribute macro `async_trait` (in Nightly builds, run with -Z macro-backtrace for more info)
```

I expected to see this happen:

Nothing.

### Version

```text
nightly-aarch64-unknown-linux-gnu installed - rustc 1.99.0-nightly (771916f90 2026-08-08)
```
